utils/iio_attr.c

```diff
@@ -129,12 +132,12 @@ static int dump_device_attributes(const struct iio_device *dev,
 		}
 	}
 	if (wbuf) {
-		gen_function("device", "dev", attr, wbuf);
-		ret = iio_device_attr_write_string(dev, attr, wbuf);
+		gen_function(type, var, attr, wbuf);
+		ret = iio_attr_write_string(attr, wbuf);
 		if (ret > 0) {
 			if (quiet == ATTR_VERBOSE)
 				printf("wrote %li bytes to %s\n", (long)ret, attr);
```

Comment:
attr needs to be wrapped with  iio_attr_get_name

Reply:
Yes, it's strange that GCC doesn't warn about that...